client-js/app/scripts/credentials.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Credential for logging in with username and password.
+ */
+export interface Password {
+  kind: 'username+password';
+  username: string;
+  password: string;
+}
+
+/**
+ * Credential for logging in with OAuth.
+ */
+export interface OAuthToken {
+  kind: 'oauth_token';
+  clientId: string;
+  clientSecret: string;
+  accessToken: string;
+  expiresAt: number;
+  idToken: string;
+  refreshToken?: string;
+}
+
+export type Credential = Password | OAuthToken;
+
+/**
+ * Load credential from the given Storage object (session or local).
+ *
+ * If valid a credential is not found, return null.
+ */
+export function loadCredential(manager: string, storage: Storage): Credential {
+  var key = 'labrad-credentials';
+  if (manager) {
```

Comment:
I guess the compiler should catch anything non-string going into this, so a more formal `if` would be redundant.